crates/edr_napi/src/solidity_tests/artifact.rs

```diff
@@ -57,41 +63,94 @@ pub struct ContractData {
+            abi: Some(serde_json::from_str(&contract.abi).map_err(|_err| {
                 napi::Error::new(napi::Status::GenericFailure, "Invalid JSON ABI")
-            })?,
-            bytecode: contract
-                .bytecode
-                .map(|b| {
-                    b.parse().map_err(|_err| {
-                        napi::Error::new(
-                            napi::Status::GenericFailure,
-                            "Invalid hex bytecode for contract",
-                        )
-                    })
-                })
-                .transpose()?,
-            deployed_bytecode: contract
-                .deployed_bytecode
-                .map(|b| {
-                    b.parse().map_err(|_err| {
-                        napi::Error::new(
-                            napi::Status::GenericFailure,
-                            "Invalid hex deployed bytecode for contract",
-                        )
-                    })
+            })?),
+            bytecode: contract.bytecode.map(|bytecode| {
+                foundry_compilers::artifacts::CompactBytecode {
+                    object: foundry_compilers::artifacts::BytecodeObject::Unlinked(bytecode),
+                    source_map: None,
+                    link_references: convert_link_references(
+                        contract.link_references.unwrap_or_default(),
+                    ),
+                }
+            }),
+            deployed_bytecode: contract.deployed_bytecode.map(|deployed_bytecode| {
+                let compact_bytecode = foundry_compilers::artifacts::CompactBytecode {
+                    object: foundry_compilers::artifacts::BytecodeObject::Unlinked(
+                        deployed_bytecode,
+                    ),
+                    source_map: None,
+                    link_references: convert_link_references(
+                        contract.deployed_link_references.unwrap_or_default(),
+                    ),
+                };
+                foundry_compilers::artifacts::CompactDeployedBytecode {
+                    bytecode: Some(compact_bytecode),
+                    immutable_references: BTreeMap::default(),
+                }
+            }),
+        })
+    }
+}
+
+impl TryFrom<ContractData> for foundry_compilers::artifacts::CompactContractBytecodeCow<'static> {
+    type Error = napi::Error;
+
+    fn try_from(contract: ContractData) -> napi::Result<Self> {
+        let c: foundry_compilers::artifacts::CompactContractBytecode = contract.try_into()?;
+        Ok(CompactContractBytecodeCow {
+            abi: c.abi.map(Cow::Owned),
+            bytecode: c.bytecode.map(Cow::Owned),
+            deployed_bytecode: c.deployed_bytecode.map(Cow::Owned),
+        })
+    }
+}
+
+fn convert_link_references(
+    link_references: HashMap<String, HashMap<String, Vec<LinkReference>>>,
+) -> BTreeMap<String, BTreeMap<String, Vec<foundry_compilers::artifacts::Offsets>>> {
```

Comment:
Do you need the entries to be sorted - since you're using a `BTreeMap`?

Reply:
Hey thanks for asking! Added a comment to the code that hopefully clarifies it:

> The order of link references as supplied through the NAPI interface doesn't matter, but the
> order can matter downstream for deterministic address generation.

There is no native support for `BTreeMap` in `napi-rs` afaik, hence the `HashMap` argument.